File: ALittle/Module/ALittleIDE/Other/GameLibrary/CEngine/Script/CEngine/Utility/ConfigSystem.lua
-- ALittle Generate Lua
module("ALittle", package.seeall)

local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittle.IFileLoader, " extends class:ALittle.IFileLoader is nil")
LuaClientFileLoader = Lua.Class(ALittle.IFileLoader, "ALittle.LuaClientFileLoader")

function LuaClientFileLoader:Load(file_path)
	local file = __CPPAPILocalFile()
	file:SetPath(file_path)
	if not file:Load() then
		return nil
	end
	local content = file:GetContent()
	file:Clear()
	return content
end

assert(ALittle.IFileSaver, " extends class:ALittle.IFileSaver is nil")
LuaClientFileSaver = Lua.Class(ALittle.IFileSaver, "ALittle.LuaClientFileSaver")

function LuaClientFileSaver:Save(file_path, content)
	return File_SaveFile(file_path, content, -1)
end

function CreateConfigSystem(file_path, print_error)
	return Lua.Template(JsonConfig, "ALittle.JsonConfig<ALittle.LuaClientFileLoader, ALittle.LuaClientFileSaver>", LuaClientFileLoader, LuaClientFileSaver)(file_path, print_error)
end

File: ALittleIDE/Module/ALittleIDE/Other/GameLibrary/CEngine/JSScript/CEngine/Utility/ConfigSystem.js
{
if (typeof ALittle === "undefined") window.ALittle = {};


if (ALittle.IFileLoader === undefined) throw new Error(" extends class:ALittle.IFileLoader is undefined");
ALittle.JClientFileLoader = JavaScript.Class(ALittle.IFileLoader, {
	Load : function(file_path) {
		let content = undefined;
		let json = undefined;
		if (document.cookie !== undefined && document.cookie !== "") {
			let error = undefined;
			[error, json] = (function() { try { let ___VALUE = ALittle.String_JsonDecode.call(undefined, document.cookie); return [undefined, ___VALUE]; } catch (___ERROR) { return [___ERROR.message]; } }).call(this);
			if (error !== undefined) {
				json = undefined;
			}
		}
		if (json === undefined) {
			return undefined;
		}
		return json[file_path];
	},
}, "ALittle.JClientFileLoader");

if (ALittle.IFileSaver === undefined) throw new Error(" extends class:ALittle.IFileSaver is undefined");
ALittle.JClientFileSaver = JavaScript.Class(ALittle.IFileSaver, {
	Save : function(file_path, content) {
		if (window.wx !== undefined) {
			return false;
		}
		let json = undefined;
		if (document.cookie !== undefined && document.cookie !== "") {
			let error = undefined;
			[error, json] = (function() { try { let ___VALUE = ALittle.String_JsonDecode.call(undefined, document.cookie); return [undefined, ___VALUE]; } catch (___ERROR) { return [___ERROR.message]; } }).call(this);
			if (error !== undefined) {
				json = undefined;
			}
		}
		if (json === undefined) {
			json = {};
		}
		json[file_path] = content;
		document.cookie = ALittle.String_JsonEncode(json);
		return true;
	},
}, "ALittle.JClientFileSaver");

ALittle.CreateConfigSystem = function(file_path, print_error) {
	return ALittle.NewObject(JavaScript.Template(ALittle.JsonConfig, "ALittle.JsonConfig<ALittle.JClientFileLoader, ALittle.JClientFileSaver>", ALittle.JClientFileLoader, ALittle.JClientFileSaver), file_path, print_error);
}

}